engine/src/iutil.cpp

```diff
@@ -977,14 +977,14 @@ bool MCImageRotateRotate(MCImageBitmap *p_src, real64_t p_angle, uint32_t p_back
 
 			if (quality == INTERPOLATION_BOX)
 			{
-				if (t_ix < 0 || t_ix >= p_src->width || t_iy < 0 || t_iy >= p_src->height)
+				if (t_ix < 0 || (uint32_t)t_ix >= p_src->width || t_iy < 0 || (uint32_t)t_iy >= p_src->height)
 					t_pixel = p_backing_color;
 				else
 					t_pixel = MCImageBitmapGetPixel(p_src, t_ix, t_iy);
 			}
 			else
 			{
-				if (t_ix < -1 || t_ix >= p_src->width || t_iy < -1 || t_iy >= p_src->height)
+				if (t_ix < -1 || (uint32_t)t_ix >= p_src->width || t_iy < -1 || (uint32_t)t_iy >= p_src->height)
```

Comment:
Unfortunately this code appears to have been _originally_ written incorrectly.

1. If `t_ix == -1`, then `uint32_t(t_ix) == UINT32_MAX`.
2. Therefore if `t_ix == -1`, then `uint32_t(t_ix) >= p_src->width` for all values of `p_src->width`
3. Therefore if `t_ix == -1` then the true case of this `if` statement is _always_ executed.

However, there are `if t_ix == -1` checks in the false case of this `if` statement, which is tautological.

It's very unclear how to make this code correct.